### PR TITLE
UIIN-2194 Move Inventory "Browse" logic into separate page: filters and result

### PR DESCRIPTION
**Pull request by usavkov-epam**
## Purpose

https://issues.folio.org/browse/UIIN-2194
Add filters and results for separate Browse page
## Approach

- Add component to display `Browse` filters including search form (`<BrowseInventoryFilters>`). Re-use existing components to display filter facets;
- Define new labels for browse options (E.g. "Browse subjects" -> "Subjects");
- Validate query before applying (E.g. query for browse contributors must NOT contain asterisks);
- Add react hook to fetch data base on filters and search query (`<useInventoryBrowse>`);
- Add component to show browse results (`<BrowseResultsList>`). Use current approach to interact with table rows (Navigate to `Search` page with prepared query);

**NOTES:**

1. Browse pagination will be implemented separately (https://issues.folio.org/browse/UIIN-2202)
2. Unit tests for new code will be added in separate PR (https://issues.folio.org/browse/UIIN-2203)

<details>
  <summary>Preview</summary>
  
chrome_ktAz5PtIQe

</details>

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [x] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [x] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

**Comment by usavkov-epam** [comment]
@folio-org/fe-tl-reviewers review it please, thanks!

File: src/components/BrowseInventoryFilters/BrowseInventoryFilters.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    onFetchFacets: fetchFacets(data),
+    parentResources: resources,
+    query: {
+      query: activeFilters.query,
+      filters: parseFiltersToStr(omit(activeFilters, ['qindex', 'query'])),
+    }
+  };
+
+  return (
+    <>
```
Comment (NikitaSedyx) [inline src/components/BrowseInventoryFilters/BrowseInventoryFilters.js:34]: fragment is not required

File: src/components/BrowseInventoryFilters/BrowseInventoryFilters.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        data={filtersData}
+        onChange={({ name, values }) => applyFilters(name, values)}
+        onClear={(name) => applyFilters(name, [])}
+      />
+    </>
+  );
+};
+
+BrowseInventoryFilters.manifest = Object.freeze({
+  query: buildManifestObject().query,
```
Comment (NikitaSedyx) [inline src/components/BrowseInventoryFilters/BrowseInventoryFilters.js:46]: do we really need stripes-connect here?

Comment (usavkov-epam) [reply src/components/BrowseInventoryFilters/BrowseInventoryFilters.js:46]: Yes, to display filters facets we should provide `query` resource for existing <InstanceFiltersBrowse> component

File: src/components/BrowseInventoryFilters/BrowseInventoryFilters.js (hunk 1 above)
Comment (usavkov-epam) [reply src/components/BrowseInventoryFilters/BrowseInventoryFilters.js]: removed

File: src/hooks/useInventoryBrowse/useInventoryBrowse.js [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+    query: buildQuery({
+      query: searchQuery || undefinedAsString,
+      qindex,
+      ...otherFilters,
+    }),
+  };
+
+  const updatePage = useCallback((_askAmount, _index, _firstIndex, direction) => {
+    // TODO: implement prev-next pagination
+    console.log('direction', direction);
```
Comment (uladislausamets) [inline src/hooks/useInventoryBrowse/useInventoryBrowse.js:89]: 👀

**Review by uladislausamets — APPROVED** [review]

**Review by vashjs — APPROVED** [review]

File: src/views/BrowseInventory/BrowseInventory.js [hunk 4]
```diff
@@ -6,18 +10,75 @@ import {
...
+    filters,
+    searchIndex,
+  });
+
+  const { validateDataQuery } = useBrowseValidation(searchIndex);
+
+  const searchableIndexesPlaceholder = intl.formatMessage({ id: 'ui-inventory.browse.searchableIndexesPlaceholder' });
+  const isResetButtonDisabled = !location.search && !searchQuery;
+
+  const formattedSearchableIndexes = useMemo(() => (
```
Comment (vashjs) [inline src/views/BrowseInventory/BrowseInventory.js:64]: I'm not sure that useMemo is required here. `browseInstanceIndexes` is static array with 3 elements, and memoization in this case will cost more than additional calculations

**Review by vashjs — APPROVED** [review]

File: src/views/BrowseInventory/BrowseInventory.js (hunk 4 above)
Comment (usavkov-epam) [reply src/views/BrowseInventory/BrowseInventory.js]: Thanks! Looks like you're right, made updates.

**Review by mkuklis — APPROVED** [review]

File: src/components/BrowseResultsList/getBrowseResultsFormatter.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+      if (r?.instance || r?.totalRecords) {
+        return getFullMatchRecord(r?.fullCallNumber, r.isAnchor);
+      }
+      return <MissedMatchItem query={r.fullCallNumber} />;
+    },
+    'contributor': r => {
+      if (r?.totalRecords) {
+        return getFullMatchRecord(r?.name, r.isAnchor);
+      }
+      return <MissedMatchItem query={r.name} />;
```
Comment (Dmytro-Melnyshyn) [inline src/components/BrowseResultsList/getBrowseResultsFormatter.js:30]: Contributor cell has already been updated [here](https://github.com/folio-org/ui-inventory/commit/cf25f8ebce1fe18299cf158eaf96fa3a3c2a4998).

Comment (usavkov-epam) [reply src/components/BrowseResultsList/getBrowseResultsFormatter.js:30]: Thanks @Dmytro-Melnyshyn, but looks like these are fresh changes and were not part of the existing implementation. This PR is not final for browse refactoring, so this and other differences could be updated in next PRs.

**Pull request merged**